### PR TITLE
Subbasin: Use Continuous Color Ramp for Streams, Catchment Load Rates

### PR DESCRIPTION
**Pull request by rajadain**
## Overview

Previously we used discrete color breaks for loading rates and concentration rates for catchments and their streams. Now we switch to a continuous color ramp where each value is interpolated within the gradient.

Each gradient is expected to have 5 stops. Beyond this, we add an addition sixth stop, immediately after the fifth, which goes from a darker red eventually to black.

This color ramp uses the same colors used for stream coloring in _variables.scss.

Connects #2827 

## Testing Instructions

* Check out this branch and `bundle`
* Go to subbasin mode and enter a HUC-12
* Turn on a related layer and observe the colors. Ensure they match with the ramp specifications in #2827.

### CODE REVIEW

File: src/mmw/js/src/modeling/models.js [hunk 1]
```diff
@@ -877,42 +878,18 @@ var SubbasinCatchmentDetailModel = Backbone.Model.extend({
...
-                    [ 250, '#A0A0A0'],
-                    [ 500, '#888888'],
-                    [ 750, '#707070'],
-                    [1000, '#484848'],
-                    [Number.MAX_VALUE, '#202020']
-                ]
+            ramps = {
+                TotalN: makeColorRamp([0, 2, 5, 10, 20]),
+                TotalP: makeColorRamp([0, 0.2, 0.5, 1.0, 2.0]),
+                Sediment: makeColorRamp([0, 200, 500, 1000, 2000]),
```
Comment (caseycesari) [inline src/mmw/js/src/modeling/models.js]: Double-checking that we don't want to use `Number.MAX_VALUE` anymore.

File: src/mmw/js/src/modeling/models.js [hunk 2]
```diff
@@ -935,29 +912,10 @@ var SubbasinCatchmentDetailModel = Backbone.Model.extend({
...
-                    [ 50, '#1A9641'],
-                    [100, '#A6D96A'],
-                    [150, '#FFFFBF'],
-                    [200, '#FDAE61'],
-                    [Number.MAX_VALUE, '#D7191C']
-                ]
+            ramps = {
+                TotalN: makeColorRamp([0, 1, 3, 6, 12]),
+                TotalP: makeColorRamp([0, 0.08, 0.2, 0.5, 1.0]),
+                Sediment: makeColorRamp([0, 8, 20, 50, 150]),
```
Comment (caseycesari) [inline src/mmw/js/src/modeling/models.js]: These numbers are different than the values they are replacing.

**Comment by rajadain** [comment]
Re: Legend, I've made a note of it in #2821. We'll need input from Design for where to put it.

**Comment by rajadain** [comment]
@aufdenkampe this work is now available on staging: https://staging.app.wikiwatershed.org/

**Comment by aufdenkampe** [comment]
@rajadain, thanks for letting me know. It's fun to play with these!
Here are a few suggestions:
* The border lines for the NHDplus catchements are too thick, interfering with the visualization of the stream segments. Please make the border lines to always be less thick than the streams (or at least no thicker than thinest stream). If there is a concern that these lines might barely be visible, that is OK with me, as the interactivity with the map should be enough to distinguish catchments.
* The darkest grey color isn't really working for me. It should be clearly Red in hue, in a ramp that shades from light red to dark red as loading rates increase. Also, there appear to be two different blocks of grey in the color ramp, one that is at the highest levels and another at moderate levels below orange.
* The implemented color ramp doesn't really make sense visually, and it doesn't seem like what I had suggested. Is there a bug? When I sort the table by loading rates, what I can discern is this ramp from low to high:
  * light green
  * salmon pink
  * green yellow? (hard to distinguish from light green)
  * beige
  * dark grey
  * grey orange with a hint of violet
  * dark grey (I can't distinguish from above)
* It will be good to have the transparency slider wired up, so that it is easier to see the colors over the base map, if desired.

**Pull request merged**